extensions/src/org/pentaho/platform/web/http/filters/PentahoPathDecodingFilter.java

```diff
@@ -40,8 +75,39 @@ public void destroy() {
   }
 
   private static final class EncodingAwareHttpServletRequestWrapper extends HttpServletRequestWrapper {
+
+    StringKeyStringValueIgnoreCaseMultivaluedMap parameterMap = null;
```

Comment:
I believe variables here will be extremely bad - this is a static final class, and every page runs though this filter. So these variables would have to minimally be threadlocal variables as one request after another would blast these things. This isn't threadsafe in any way that I can see.